### DISCUSSION TITLE
Is there a shortcut for switching tabs

### DISCUSSION THREAD

**Discussion by michaelessiet** [zed-industries/zed#6903] [Feature Requests]
Hi, so I've been looking for the keymapping for switching tabs. Is there one or is this yet to be implemented?

**Comment by SomeoneToIgnore** [comment]
Currently, there's `ctrl-Number` for opening to a tab #1/2/3/etc. and `cmd-alt-left/right arrow` to navigate to previous/next tab.
Those seem to be absent fromt the default keymap json, so there's definitely a room for improvement in the docs and/or the settings.

**Reply by verhovsky** [reply]
It should be ctrl-tab and ctrl-shift-tab with VS Code bindings. I also tried searching for a "next tab" command to try to find the shortcut

**Comment by avi-cenna** [comment]
For anyone looking for the `ctrl-tab` bindings, put this in your keymap. (To open the keymap, press `cmd-k cmd-s`)

```json
[
  {
    "context": "Pane",
    "bindings": {
      "ctrl-tab": "pane::ActivateNextItem",
      "ctrl-shift-tab": "pane::ActivatePrevItem",
    }
  }
]
```

**Reply by verhovsky** [reply]
On VS Code ctrl-tab works like alt-tab, it goes to the most recently open tab and continues through the history.

**Reply by cshaa** [reply]
This is exactly what I was looking for, thank you!

**Reply by cesc1989** [reply]
So many thanks.

**Reply by ackmanx** [reply]
Brilliant! I feel at home now switching between Intellij and Zed

**Reply by adsick** [reply]
I find this one confusing because OP asked about tabs and this are panes ("splits"), not tabs.

**Comment by ghost** [comment]
Another binding that works is `cmd+shift+[` or `cmd+shift+]` to go to the previous or next tabs.

**Reply by soupy0x** [reply]
this is the way 🥷

**Comment by rommyarb** [comment]
I tried `"ctrl-tab": ["tab_switcher::Toggle", { "select_last": true }],` but it doesn't work 🫤
I got it from here: [default-macos.json](https://github.com/zed-industries/zed/blob/main/assets/keymaps/default-macos.json)

**Reply by ghost** [reply]
Hey @rommyarb, I was wondering the same thing and realized the new tab switcher is in the preview release and has not yet made it to an official release.

See release notes for `0.130.0-pre` https://github.com/zed-industries/zed/releases/tag/v0.130.0-pre:
> Added a tab switcher, which is accessible via ctrl-tab and ctrl-shift-tab (https://github.com/zed-industries/zed/issues/7321, https://github.com/zed-industries/zed/issues/7653). The tab switcher works in both the main workspace area and in the terminal panel.

**Reply by rommyarb** [reply]
@pseudomata Ah yes, you are right. I have installed the preview release and I got it working now. Thanks!

**Reply by alygin** [reply]
> I tried "ctrl-tab": ["tab_switcher::Toggle", { "select_last": true }], but it doesn't work 🫤
> I got it from here: [default-macos.json](https://github.com/zed-industries/zed/blob/main/assets/keymaps/default-macos.json)

@rommyarb, be careful, this `select_last: true` makes tab switcher start with the last item selected. This behaviour is mapped to `ctrl-shift-tab` by default, while `ctrl-tab` automatically selects the second list item. Given that tabs in the switcher are sorted by reverse activation order, this allows the user to switch to the previously used tab with a single keystroke.

Your custom configuration breaks this feature.

**Comment by cyruzin** [comment]
Update:

```
  {
      "context": "Pane",
      "bindings": {
        "ctrl-tab": "pane::ActivateNextItem",
         // this one changed from "ActivatePrevItem" to "ActivatePreviousItem"
        "ctrl-shift-tab": "pane::ActivatePreviousItem",
      }
    }
```
